compiler/luci/service/src/Nodes/CircleReshape.cpp

```diff
@@ -154,7 +153,14 @@ loco::TensorShape Algorithm::visit(const luci::CircleReshape *node)
 
     for (uint32_t axis = 0; axis < shape_by_attr.rank(); ++axis)
     {
-      shape_by_attr.dim(axis) = node->newShape()->dim(axis);
+      if (node->newShape()->dim(axis) > 0)
+      {
+        shape_by_attr.dim(axis) = node->newShape()->dim(axis);
+      }
+      else
+      {
+        shape_by_attr.dim(axis).unset(); // unset means unknown dimension
+      }
```

Comment:
how is this change related with removing `shape_dummy` ?

Reply:
For my model converted from Pytorch I would like to [use dimensions stored in newShape](https://github.com/Samsung/ONE/blob/e5e151d547616f1ad3be6ff13f86b53990b7c157/compiler/luci/service/src/Nodes/CircleReshape.cpp#L126-L137) but models converted from Pytorch have no `luci::CircleOutputDummy`

Reply:
The purpose of the change is to align way of dynamic shape representation: `-1` for `newShape` and `0` for `shape_by_attr`.
But relation with `shape_dummy` removing is only to not show [info log about shape mismatch](https://github.com/Samsung/ONE/blob/e5e151d547616f1ad3be6ff13f86b53990b7c157/compiler/luci/service/src/Nodes/CircleReshape.cpp#L163-L165).

Reply:
> is only to not show [info log about shape mismatch](https://github.com/Samsung/ONE/blob/e5e151d547616f1ad3be6ff13f86b53990b7c157/compiler/luci/service/src/Nodes/CircleReshape.cpp#L163-L165).

I don't understand this. plz explain your understandings about this.

Reply:
Let's consider even my case from `reshape_by_newShape_dynamic` unit test:
`shape_by_input` here after https://github.com/Samsung/ONE/blob/e5e151d547616f1ad3be6ff13f86b53990b7c157/compiler/luci/service/src/Nodes/CircleReshape.cpp#L130-L135 is `[0,2,0]`
while `shape_by_attr` is `[-1,2,-1]`
`TensorShape::operator==` doesn't handle -1 as dynamic dimension

Reply:
>  doesn't handle -1 as dynamic dimension

Not this. I'm asking about the relation with `shape_dummy ` removal.
I understand about handling 0 to be handled as unknown.

Reply:
And I really don't understand why you are mentioning about the log.
What is the problem with the log?